internal/storage/driver/memory/binding.go

```diff
@@ -32,14 +32,26 @@ func (s *Binding) Insert(binding *internal.Binding) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	if _, found := s.data[binding.ID]; found {
+	if foundBinding, found := s.data[binding.ID]; found && binding.InstanceID == foundBinding.InstanceID {
 		return dberr.AlreadyExists("binding with id %s already exist", binding.ID)
 	}
 	s.data[binding.ID] = *binding
 
 	return nil
 }
 
+func (s *Binding) Update(binding *internal.Binding) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if foundBinding, found := s.data[binding.ID]; !found || binding.InstanceID != foundBinding.InstanceID {
```

Comment:
```suggestion
	if foundBinding, found := s.data[binding.ID]; !(found && binding.InstanceID == foundBinding.InstanceID) {
```
I am not sure which is better.

Reply:
First is definitely shorter.